### PR TITLE
Edit challenge view

### PR DESCRIPTION
**Pull request by JMurphyWeb**
This pr works off of https://github.com/emfoundation/ce100-app/pull/710  (add challenge pr)

@harrygfox @mk4111 Sorry it is a bit generic, but there was a lot of overlapp - this PR is a bit bigger, but I would recommend reviewing this PR instead of the above one as it contains a refactor of the form elements & modifiers.

- Brings together all the form elements and modifiers (documented here https://github.com/emfoundation/ce100-app/issues/711)
- Adds tags to the edit challenge view
- Fixes bottom navbar

Edit challenge issue: #693 

Add challenge issue: #692 

From the previous PR this one also covers:

- input fields (text and textarea)
- labels and spacing between them
- form copy
- form sections (lines separating content)

### CODE REVIEW

File: templates/views/browse/categories-tags.html [hunk 1]
```diff
@@ -10,7 +10,7 @@ <h1 class="title">Select Tags</h1>
     <form class="form form--tags" method="post">
         <div class="form__section">
           <label class="tag__title">
-            <a class="anchor--danger text-decoration" href="/{{pageType}}">Search all tags</a>
+            <a class="anchor--red text-decoration" href="/{{pageType}}">Search all tags</a>
```
Comment (harrygfox) [inline templates/views/browse/categories-tags.html]: text-decoration appears to be some sort of utility class!

**Comment by JMurphyWeb** [comment]
@harrygfox will look into it. Thanks!

**Comment by JMurphyWeb** [comment]
@harrygfox it is fixed in the following PR: https://github.com/emfoundation/ce100-app/pull/715/files#diff-37bef4d5c516db177d3f7205df749e21R59

File: templates/views/browse/categories-tags.html (hunk 1 above)
Comment (JMurphyWeb) [inline templates/views/browse/categories-tags.html]: I will address this in the next PR: https://github.com/emfoundation/ce100-app/pull/715

Comment (JMurphyWeb) [inline templates/views/browse/categories-tags.html]: sorted now in said PR.

**Comment by harrygfox** [comment]
@JMurphyWeb Fandabbydozy. So this merge affected #710 which has also been auto-merged. Over to you for please-test labels etc

**Pull request merged**